### ISSUE TITLE
Create root cluster visualizer (t-SNE/UMAP)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#436]
## Goal

Generate 2D visualizations of root embeddings using t-SNE or UMAP to show semantic clusters.

## Use Cases

- Visualize semantic groupings (animals, family, cognitive, etc.)
- Find outliers or poorly trained embeddings
- Validate semantic structure
- Create figures for documentation/presentations
- Debug embedding quality

## Implementation

Create `scripts/visualize_root_clusters.py`:

```python
#!/usr/bin/env python3
"""Visualize root embeddings in 2D using t-SNE or UMAP."""

import torch
from sklearn.manifold import TSNE
import matplotlib.pyplot as plt

def visualize_embeddings(model_path, method='tsne', sample_size=500):
    # Load embeddings
    # Sample roots (or use all if < 1000)
    # Apply dimensionality reduction
    # Plot with labels for interesting roots
    # Color by semantic category if available
    pass
```

## Features

- [ ] t-SNE visualization
- [ ] UMAP visualization (optional, requires umap-learn)
- [ ] Interactive plot (plotly)
- [ ] Highlight semantic groups (animals, family, etc.)
- [ ] Label high-frequency/Fundamento roots
- [ ] Export as PNG/SVG

## Example Output

```bash
python scripts/visualize_root_clusters.py
# Creates: visualizations/root_embeddings_tsne.png

python scripts/visualize_root_clusters.py --method umap --interactive
# Opens: interactive HTML plot
```

## Success Criteria

- Clear semantic clusters visible
- Similar roots cluster together (hund, kat, bird)
- Different semantic fields separate
- Fundamento roots well-distributed

## Deliverables

- [ ] `scripts/visualize_root_clusters.py`
- [ ] Example visualizations in `docs/` or `.wiki/`
- [ ] Optional: Animated visualization showing training progression

## Dependencies

```
scikit-learn  # Already in requirements
matplotlib    # Already in requirements
plotly        # Optional for interactive
umap-learn    # Optional alternative to t-SNE
```

## Related

- #427 - Root embeddings validation
- #430 - Root similarity query tool
